python_modules/dagster/dagster/_core/definitions/metadata/source_code.py

```diff
@@ -142,14 +142,36 @@ def _with_code_source_single_definition(
     )
 
 
+class SourceControlFilePathMapping(NamedTuple):
+    """Specifies the mapping between local file paths and their corresponding paths in a source control repository,
+    using a specific anchor file as a reference point. All other paths are calculated relative to this anchor file.
+
+    Args:
+        local_file_anchor (Path): The path to a local file that is present in the repository.
+        file_anchor_path_in_repository (Path): The path to the anchor file in the repository.
+
+    Example:
+        .. code-block:: python
+
+            anchor = SourceControlFilePathMapping(
+                local_file_anchor=Path(__file__),
+                file_anchor_path_in_repository="python_modules/my_module/my-module/__init__.py",
+            )
+    """
+
+    local_file_anchor: Path
+    file_anchor_path_in_repository: str
```

Comment:
one is a string and the other is a path?

Reply:
It feels slightly wrong to use `Path` to represent a "remote" path, e.g. one not on the local filesystem, since `Path` uses the local file delimiter (windows vs unix), but maybe I'm overthinking things. I think we can force it to be a unix-style path by wrapping a user-passed `Path` in a `PurePosixPath` .